spec/models/use_cases/create_diagnosis_spec.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe UseCases::CreateDiagnosis do
+  describe 'create_for_params' do
+    context 'no selected need' do
+      let(:visit) { create :visit, id: 4 }
+      let(:params) { { 'visit_id' => visit.id.to_s } }
+
+      before { described_class.create_with_params params }
+
+      it 'creates a diagnosis linked to the right visit' do
```

Comment:
No need to have `let` and `before`, add everything in `it`.
You can add comments such as `# Given` `# When` `# Then` if you want to explicitly separate these lines.